### PR TITLE
TLS functional tests

### PR DESCRIPTION
**Pull request by gulducat**
Extend functional tests to also run with TLS enabled on Consul, Nomad, and Vault.

Recommend reviewing `functional_test.go` [ignoring whitespace](https://github.com/hashicorp/hcdiag/pull/110/files?diff=unified&w=1).  The diff looks wonky with most of the test content being indented, but the actual changes are pretty minimal.

Example test error output if the selected TLS-error indicator string ("x509:") is found in hcdiag command output:

```
functional_test.go:117: 
            Error Trace:    functional_test.go:168
                                                    functional_test.go:117
            Error:          "2022-05-06T11:03:32.413-0500 [WARN]  hcdiag: result: seeker="GET /v1/agent/self" result=%!s(<nil>) error="Get "https://127.0.0.1:8501/v1/agent/self": x509: “hcdiag-test” certificate is not trusted"" should not contain "x509:"
            Test:           TestFunctional/tls/consul
            Messages:       unexpected TLS error
functional_test.go:117: 
            Error Trace:    functional_test.go:168
                                                    functional_test.go:117
            Error:          "2022-05-06T11:03:32.417-0500 [WARN]  hcdiag: result: seeker="GET /v1/agent/metrics" result=%!s(<nil>) error="Get "https://127.0.0.1:8501/v1/agent/metrics": x509: “hcdiag-test” certificate is not trusted"" should not contain "x509:"
            Test:           TestFunctional/tls/consul
            Messages:       unexpected TLS error
```

### CODE REVIEW

**Review by gulducat — COMMENTED** [review]
am I able to trigger functional test by adding a review to my own PR? 🤔

File: tests/integration/certs/tls.tf [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# this is here to conveniently generate a self-signed CA and use it to sign a cert/key pair.
+
+locals {
+  valid_hours = 24 * 365 * 50 // 50 years
+
+  files = {
+    # "ca.key"     = tls_private_key.ca.private_key_pem
```
Comment (nwchandler) [inline tests/integration/certs/tls.tf:7]: Should we remove this commented-out line?

**Review by nwchandler — APPROVED** [review]
One small comment inline, but otherwise LGTM

File: tests/integration/certs/tls.tf (hunk 1 above)
Comment (gulducat) [reply tests/integration/certs/tls.tf]: you got it!  I just amended that commit for tidy git history ⚡

**Pull request merged**